app/routes/_auth+/signup.tsx

```diff
@@ -64,24 +64,25 @@ export async function loader({ request }: DataFunctionArgs) {
 			},
 		})
 	}
-	return json({ fields: getFieldsFromSchema(signupSchema) })
+	return json({})
 }
 
 export async function action({ request }: DataFunctionArgs) {
 	const formData = await request.formData()
-	const result = await signupSchema.safeParseAsync(
-		preprocessFormData(formData, signupSchema),
-	)
-	if (!result.success) {
-		return json(
-			{
-				status: 'error',
-				errors: result.error.flatten(),
-			} as const,
-			{ status: 400 },
-		)
+	const submission = parse(formData, {
+		schema: signupSchema,
+		acceptMultipleErrors: () => true,
+	})
+	if (!submission.value) {
+		return json({
+			status: 'error',
+			submission,
+		} as const)
 	}
-	const { email } = result.data
+	if (submission.intent !== 'submit') {
+		return json({ status: 'success', submission } as const)
+	}
```

Comment:
Same here regarding removing this if statement and updating the previous one.

Reply:
Done.